spec/lib/lighthouse/facilities/client_spec.rb

```diff
@@ -0,0 +1,113 @@
+        'fax' => '632-310-5962',
+        'main' => '632-550-3888',
+        'patient_advocate' => '632-550-3888 x3716',
+        'pharmacy' => '632-550-3888 x5029'
+      },
+      hours: {
+        'friday' => '730AM-430PM',
+        'monday' => '730AM-430PM',
+        'saturday' => 'Closed',
+        'sunday' => 'Closed',
+        'thursday' => '730AM-430PM',
+        'tuesday' => '730AM-430PM',
+        'wednesday' => '730AM-430PM'
+      },
+      services: { 'health' => %w[Audiology Cardiology Dermatology EmergencyCare Gastroenterology
+                                 MentalHealthCare Ophthalmology Orthopedics PrimaryCare SpecialtyCare],
+                  'last_updated' => '2020-03-30', 'other' => [] },
+      feedback: {
+        'effective_date' => '2019-06-20',
+        'health' => { 'specialty_care_routine' => 0.9100000262260437 }
+      },
+      access: {
+        'effective_date' => '2020-03-30',
+        'health' => [
+          { 'established' => 25.0, 'new' => 139.0, 'service' => 'Audiology' },
+          { 'established' => 34.944444, 'new' => 66.0, 'service' => 'Cardiology' },
+          { 'established' => 2.583333, 'new' => 103.5, 'service' => 'Dermatology' },
+          { 'established' => nil, 'new' => 104.0, 'service' => 'Gastroenterology' },
+          { 'established' => 26.101351, 'new' => 103.333333, 'service' => 'MentalHealthCare' },
+          { 'established' => 12.864864, 'new' => 137.0, 'service' => 'Ophthalmology' },
+          { 'established' => 21.325, 'new' => 136.9, 'service' => 'Orthopedics' },
+          { 'established' => 16.7266, 'new' => 21.566666, 'service' => 'PrimaryCare' },
+          { 'established' => 22.189427, 'new' => 70.949367, 'service' => 'SpecialtyCare' }
+        ]
+      },
+      mobile: false,
+      active_status: 'A',
+      visn: '21',
+      operating_status: { 'code' => 'NORMAL' },
+      facility_type_prefix: 'vha',
+      unique_id: '358'
+    }
+  end
+
+  it 'is an Facilities::Client object' do
+    facilities_client = described_class.new(Settings.lighthouse.api_key)
+    expect(facilities_client).to be_an(Lighthouse::Facilities::Client)
+    expect(facilities_client).to have_attributes(headers: { 'apikey' => Settings.lighthouse.api_key })
+  end
+
+  context 'with an http timeout' do
+    it 'logs an error and raise GatewayTimeout' do
+      allow_any_instance_of(Faraday::Connection).to receive(:get).and_raise(Faraday::TimeoutError)
+      expect do
+        Lighthouse::Facilities::Client.new(Settings.lighthouse.api_key).get_facilities(params)
+      end.to raise_error(Common::Exceptions::GatewayTimeout)
+    end
+  end
+
+  describe '#get_by_id' do
+    it 'returns a facility' do
+      VCR.use_cassette('/lighthouse/facilities', match_requests_on: %i[path query]) do
+        r = Lighthouse::Facilities::Client.new(Settings.lighthouse.api_key).get_by_id('vha_358')
+        expect(r).to have_attributes(vha_358_attributes)
+      end
+    end
+  end
+
+  describe '#get_facilities' do
```

Comment:
Should we have a test for the error cases described in the documentation?

Reply:
Added tests for 400, 401. and 404 errors.